rest/model/github_hook.go

```diff
@@ -12,28 +12,19 @@ type APIGithubHook struct {
 	Repo   *string `json:"repo"`
 }
 
-func (a *APIGithubHook) BuildFromService(h interface{}) error {
-	v, ok := h.(model.GithubHook)
-	if !ok {
-		return errors.Errorf("programmatic error: expected GitHub hook but got type %T", h)
-	}
-	if v.HookID == 0 {
+func (a *APIGithubHook) BuildFromService(hook model.GithubHook) error {
```

Comment:
Should we remove the error returning here and just ensure it's caught in validation? So we can adhere to the pattern of not returning an err in `BuildFromService`